contributor-guide/modules/ROOT/pages/testing/continuous-integration.adoc

```diff
@@ -1,71 +1,94 @@
 = Continuous Integration
 
+
 The Boost project uses Continuous Integration (CI) to ensure the quality and integrity of its code. CI is the practice of merging all developers' working copies to a shared mainline several times a day. The main aim is to prevent integration problems, which are detected and can be fixed as early as possible.
 
 Boost uses several CI services for testing on different platforms and compilers. Many libraries use two or three of the systems described here, as does the https://github.com/boostorg/boost/tree/master[Super-project] itself.
 
 Note:: It is a requirement for a new library submission to Boost to include an appropriate CI system. Refer to the examples for each CI system to better understand what is involved.
 
+== New Library CI Framework
```

Comment:
I think this is missing the Boost-specific details, which are what makes CI testing in Boost difficult in my opinion. Some things that were really hard for me:
* You test with the superproject, cloning it and placing your lib inside. No-one else does this, you normally would consume Boost as a dependency.
* You test with the develop or master branches of the superproject, instead of a stable release.
* The cmake workflows are not trivial and not usual cmake.

I don't think we can expect @PeterTurcan to know these things because they are very Boost specific, as I recall Vinnie said. This is a really nice start for the CI documentation, but I think me, @alandefreitas or any other author should add the nitty-gritty details this is missing.

Reply:
These seem like valid points to make, so added them. More detail from lib authors can be added in due course.